src/charm.py

```diff
@@ -28,7 +31,7 @@ class MySQLCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
-        self._stored.set_default(mysql_setup={})
+        self._stored.set_default(mysql_setup={"MYSQL_ROOT_PASSWORD": False})
```

Comment:
Possibly intentional/a style thing, but you can set all your default StoredState values in one call:
```
self._stored.set_default(
    mysql_initialized=False,
    mysql_setup={"MYSQL_ROOT_PASSWORD": False},
) 
```

Reply:
Nice!